homeassistant/components/climate/__init__.py

```diff
@@ -239,309 +185,258 @@ def state_attributes(self):
         return None
 
     @property
-    def target_humidity(self):
+    def target_humidity(self) -> Optional[int]:
         """Return the humidity we try to reach."""
         return None
 
     @property
-    def current_operation(self):
-        """Return current operation ie. heat, cool, idle."""
-        return None
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode.
+
+        Need to be one of HVAC_MODE_*.
+        """
+        raise NotImplementedError()
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes.
+
+        Need to be a subset of HVAC_MODES.
+        """
+        raise NotImplementedError()
 
     @property
-    def operation_list(self):
-        """Return the list of available operation modes."""
+    def hvac_action(self) -> Optional[str]:
+        """Return the current running hvac operation if supported.
+
+        Need to be one of CURRENT_HVAC_*.
+        """
         return None
 
     @property
-    def current_temperature(self):
+    def current_temperature(self) -> Optional[float]:
         """Return the current temperature."""
         return None
 
     @property
-    def target_temperature(self):
+    def target_temperature(self) -> Optional[float]:
         """Return the temperature we try to reach."""
         return None
 
     @property
-    def target_temperature_step(self):
+    def target_temperature_step(self) -> Optional[float]:
         """Return the supported step of target temperature."""
         return None
 
     @property
-    def target_temperature_high(self):
+    def target_temperature_high(self) -> Optional[float]:
         """Return the highbound target temperature we try to reach."""
         return None
 
     @property
-    def target_temperature_low(self):
+    def target_temperature_low(self) -> Optional[float]:
         """Return the lowbound target temperature we try to reach."""
         return None
 
     @property
-    def is_away_mode_on(self):
-        """Return true if away mode is on."""
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
         return None
 
     @property
-    def current_hold_mode(self):
-        """Return the current hold mode, e.g., home, away, temp."""
+    def preset_modes(self) -> Optional[List[str]]:
+        """Return a list of available preset modes."""
         return None
 
     @property
-    def is_on(self):
-        """Return true if on."""
-        return None
-
-    @property
-    def is_aux_heat_on(self):
+    def is_aux_heat(self) -> Optional[str]:
         """Return true if aux heater."""
         return None
 
     @property
-    def current_fan_mode(self):
+    def fan_mode(self) -> Optional[str]:
         """Return the fan setting."""
         return None
 
     @property
-    def fan_list(self):
+    def fan_modes(self) -> Optional[List[str]]:
```

Comment:
Should we name all these "available options" properties as just the plural form of the word?

For example: `fan_mode` and `fan_modes`.

I fear this will lead to bugs because the two are very difficult to distinguish when looking over code.

Could we instead name them `_list` so for example `fan_mode_list`

Reply:
Or `fan_support` ? @balloob